event_model/schemas/event_page.json

```diff
@@ -3,61 +3,83 @@
    "description": "Page of documents to record a quanta of collected data",
    "type": "object",
    "definitions": {
-      "dataframe": {
-         "title": "dataframe",
-         "description": "A DataFrame-like object",
+      "Dataframe": {
+         "title": "Dataframe",
          "type": "object",
          "additionalProperties": {
-            "type": "array"
+            "type": "array",
+            "items": {}
          }
       },
-      "dataframe_for_filled": {
-         "title": "dataframe_for_filled",
-         "description": "A DataFrame-like object with boolean or string entries",
+      "DataframeForFilled": {
+         "title": "DataframeForFilled",
          "type": "object",
          "additionalProperties": {
             "type": "array",
             "items": {
-               "type": [
-                  "boolean",
-                  "string"
+               "anyOf": [
+                  {
+                     "type": "boolean"
+                  },
+                  {
+                     "type": "string"
+                  }
                ]
             }
          }
       }
    },
    "properties": {
       "data": {
+         "title": "Data",
          "description": "The actual measurement data",
-         "$ref": "#/definitions/dataframe"
+         "allOf": [
+            {
+               "$ref": "#/definitions/Dataframe"
+            }
+         ]
       },
       "descriptor": {
+         "title": "Descriptor",
          "description": "The UID of the EventDescriptor to which all of the Events in this page belong",
          "type": "string"
       },
       "filled": {
+         "title": "Filled",
          "description": "Mapping each of the keys of externally-stored data to an array containing the boolean False, indicating that the data has not been loaded, or to foreign keys (moved here from 'data' when the data was loaded)",
-         "$ref": "#/definitions/dataframe_for_filled"
+         "allOf": [
+            {
+               "$ref": "#/definitions/DataframeForFilled"
+            }
+         ]
```

Comment:
Is there any way to make pydantic put in just `"#/definitions/dataframe_for_filled"` rather than `"allof"...`?

Reply:
@coretl 

Doesn't look as pretty but the behaviour is the same as if the "allOf" weren't there. This seems to be pydantic's behaviour whenever a basemodel is included as a type in another basemodel:

``` python
class X(TypedDict):
    x: Annotated[int, Field(description="x field")]

modelX = create_model(X.__name__, __config__=Config, **get_annotations(X))



class Y(TypedDict):
    x: Annotated[modelX, Field(description="x field")]
    y: Annotated[int, Field(description="y field")]

modelY = create_model(Y.__name__, __config__=Config, **get_annotations(Y))

```
gives us:

```
{'definitions': {'X': {'properties': {'x': {'description': 'x field',
                                            'title': 'X',
                                            'type': 'integer'}},
                       'required': ['x'],
                       'title': 'X',
                       'type': 'object'}},
 'properties': {'x': {'allOf': [{'$ref': '#/definitions/X'}],
                      'description': 'x field',
                      'title': 'X'},
                'y': {'description': 'y field',
                      'title': 'Y',
                      'type': 'integer'}},
 'required': ['x', 'y'],
 'title': 'Y',
 'type': 'object'}
```

You think it's worth writing another function, ran after generation, which replaces any

``` python
"allOf": [{"$ref": "#/definition/some_class"}]
```
with

``` python
"$ref": "#/definition/some_class"
```
?

Reply:
If it is functionally equivalent let's leave it like it is. It might be fixed in Pydantic 2 anyway...